tensorflow/tools/ci_build/builds/test_on_install.sh

```diff
@@ -0,0 +1,211 @@
+#
+# To select only a subset of the Python tests to run, set the environment
+# variable PY_TEST_WHITELIST, e.g.,
+#   PY_TEST_WHITELIST="tensorflow/python/kernel_tests/shape_ops_test.py"
+# Separate the tests with a colon (:). Leave this environment variable empty
+# to disable the whitelist.
+#
+# You can also ignore a set of the tests by using the environment variable
+# PY_TEST_BLACKLIST. For example, you can include in PY_TEST_BLACKLIST the
+# tests that depend on Python modules in TensorFlow source that are not
+# exported publicly.
+#
+# In addition, you can put blacklist for only GPU build inthe environment
+# variable PY_TEST_GPU_BLACKLIST.
+#
+# If the environmental variable NO_TEST_ON_INSTALL is set to any non-empty
+# value, the script will exit after the pip install step.
+
+echo "PY_TEST_BLACKLIST: ${PY_TEST_BLACKLIST}"
+echo "PY_TEST_GPU_BLACKLIST: ${PY_TEST_GPU_BLACKLIST}"
+
+# Get the command line arguments
+CONTAINER_TYPE=$( echo "$1" | tr '[:upper:]' '[:lower:]' )
+
+# Append GPU-only test blacklist
+if [[ ${CONTAINER_TYPE} == "gpu" ]]; then
+  PY_TEST_BLACKLIST="${PY_TEST_BLACKLIST}:${PY_TEST_GPU_BLACKLIST}"
+fi
+
+# Build the pip package
+PIP_BUILD_TARGET="//tensorflow/tools/pip_package:build_pip_package"
+if [[ ${CONTAINER_TYPE} == "cpu" ]]; then
+  bazel build -c opt ${PIP_BUILD_TARGET}
+elif [[ ${CONTAINER_TYPE} == "gpu" ]]; then
+  bazel build -c opt --config=cuda ${PIP_BUILD_TARGET}
+else
+  echo "Unrecognized container type: ${CONTAINER_TYPE}"
+  exit 1
+fi
+
+# Build PIP Wheel file
+RAND_STR=`cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 8 | head -n 1`
+PIP_WHL_DIR="/tmp/tensorflow_whl_${RAND_STR}"
+rm -rf ${PIP_WHL_DIR} && mkdir ${PIP_WHL_DIR}
+bazel-bin/tensorflow/tools/pip_package/build_pip_package ${PIP_WHL_DIR} &&
+
+# Perform installation
+WHL_PATH=`ls ${PIP_WHL_DIR}/tensorflow*.whl`
+if [[ `echo ${WHL_PATH} | wc -w` -ne 1 ]]; then
+  echo "ERROR: Failed to find exactly one built TensorFlow .whl file"
+  exit 1
+fi
+
+echo "whl file path = ${WHL_PATH}"
+
+# Install, in user's local home folder
+pip install --user ${WHL_PATH} &&
+
+# If NO_TEST_ON_INSTALL is set to any non-empty value, skip all Python
+# tests-on-install and exit right away
+if [[ ! -z ${NO_TEST_ON_INSTALL} ]]; then
+  echo "NO_TEST_ON_INSTALL=${NO_TEST_ON_INSTALL}:"
+  echo "  Skipping ALL Python unit tests on install"
+  exit 0
+fi
+
+# Directory from which the unit-test files will be run
+PY_TEST_DIR="/tmp/tf_py_tests_${RAND_STR}"
+rm -rf ${PY_TEST_DIR}
+mkdir ${PY_TEST_DIR}
+
+# Copy source files that are required by the tests but are not included in the
+# PIP package
+LIB_PYTHON_DIR=`ls -d ${HOME}/.local/lib/python*`
+PACKAGES_DIR=`ls -d ${LIB_PYTHON_DIR}/*-packages`
+
+echo "Copying some source directories that are required by tests but are "\
+"not included in install to Python packages directory: ${PACKAGES_DIR}"
+
+# tensorflow.python.tools
+rm -rf ${PACKAGES_DIR}/tensorflow/python/tools
+cp -r /tensorflow/tensorflow/python/tools \
```

Comment:
remove the leading /tensorflow from here and the lines bellow

if you do that people can run it on their machines outside docker